modules/opentracing/src/main/scala/OTTracer.scala

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2019 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package natchez
+package opentracing
+
+import cats.Applicative
+import cats.effect.{Bracket, ExitCase, Resource, Sync}
+import cats.implicits._
+import io.opentracing.SpanContext
+import io.opentracing.log.Fields
+import io.opentracing.propagation.{Format, TextMapAdapter}
+import io.{opentracing => ot}
+
+import scala.jdk.CollectionConverters._
+
+object OTTracer {
+
+  private [opentracing] def makeSpan[F[_] : Sync](t: ot.Tracer)(acquireSpan: F[ot.Span]): Resource[F, Span[F]] =
+    Resource.makeCase(acquireSpan)(finishSpan(t)).map(OTSpan(t, _))
+
+  private def finishSpan[F[_] : Bracket[*[_], Throwable] : Sync](t: ot.Tracer): (ot.Span, ExitCase[Throwable]) => F[Unit] =
+    (s, exitCase) => {
+      def attachPossibleException: ExitCase[Throwable] => F[Unit] = {
+        case ExitCase.Error(ex) =>
+          val map: java.util.Map[String, Throwable] = new java.util.HashMap[String, Throwable]
+          map.put(Fields.ERROR_OBJECT, ex)
+
+          Sync[F].delay(s.log(map)).void
+        case _ => Applicative[F].unit
+      }
+
+      Bracket[F, Throwable].guarantee(attachPossibleException(exitCase))(Sync[F].delay {
```

Comment:
There's syntax for `guarantee` in `cats.effect.syntax.bracket._`. Also, I think `a.attempt *> b` would be enough as `finishSpan` will run in resource cleanup, which is uncancelable.